### PR TITLE
Issue#434 [done]

### PR DESCRIPTION
**Pull request by sanket407**
Issue #434 434
merged NugetHierarchicalArtifactCoorfinates Class into NugetArtifactCoordinates Class
Refactored "Nuget Hierarchical" into "NuGet"
Refactored "NUGET_HIERARCHICAL" into "NUGET"

### CODE REVIEW

**Comment by carlspring** [comment]
Great!

Could you also  please fix the issue above reported by Sonar?

**Comment by sanket407** [comment]
SonarQube asks for login credentials

**Comment by carlspring** [comment]
The issue is reported by Sonar directly in the pull request. No need to log in to Sonar, as it has already pointed out what and where the issue is.

**Comment by sanket407** [comment]
I thought sonar more details of the error are given after logging in !
Anyways the pull seems ready to merge !

**Comment by carlspring** [comment]
Excellent! Thank you! I'll review and merge it.

File: strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-nuget-layout-provider/src/main/java/org/carlspring/strongbox/artifact/coordinates/NugetArtifactCoordinates.java [hunk 1]
```diff
@@ -1,23 +1,78 @@
...
+        {
+            return;
+        }
+
+        setId(packageId);
+        setVersion(version);
+        setType(packageArtifactType);
+    }
+
+    public String toPath()
```
Comment (sbespalov) [inline strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-nuget-layout-provider/src/main/java/org/carlspring/strongbox/artifact/coordinates/NugetArtifactCoordinates.java]: @sanket407 can you please move this method below the constructors (better to the end of the class definition)

File: strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-nuget-layout-provider/src/main/java/org/carlspring/strongbox/artifact/coordinates/NugetArtifactCoordinates.java [hunk 2]
```diff
@@ -1,23 +1,78 @@
...
+        if ("nuspec".equals(typeLocal))
+        {
+            return String.format("%s/%s/%s.%s", idLocal, versionLocal, idLocal, typeLocal);
+        }
+
+        return String.format("%s/%s/%s.%s.%s", idLocal, versionLocal, idLocal, versionLocal, typeLocal);
+    }
+
+    @Override
+    public URI toResource()
```
Comment (sbespalov) [inline strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-nuget-layout-provider/src/main/java/org/carlspring/strongbox/artifact/coordinates/NugetArtifactCoordinates.java]: same as above

File: strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-nuget-layout-provider/src/main/java/org/carlspring/strongbox/providers/layout/NugetLayoutProvider.java [hunk 3]
```diff
@@ -42,14 +43,14 @@
  *
  */
 @Component
-public class NugetHierarchicalLayoutProvider
-        extends AbstractLayoutProvider<NugetHierarchicalArtifactCoordinates,
+public class NugetLayoutProvider
+        extends AbstractLayoutProvider<NugetArtifactCoordinates,
```
Comment (sbespalov) [inline strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-nuget-layout-provider/src/main/java/org/carlspring/strongbox/providers/layout/NugetLayoutProvider.java]: please fix the indentation here

File: strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-nuget-layout-provider/src/main/java/org/carlspring/strongbox/repository/NugetRepositoryFeatures.java [hunk 4]
```diff
@@ -128,22 +127,22 @@ public boolean downloadRemoteFeed(String storageId,
...
+            Set<NugetArtifactCoordinates> artifactToSaveSet = new HashSet<>();
             for (PackageEntry packageEntry : packageFeed.getEntries())
             {
                 String packageId = packageEntry.getProperties().getId();
                 packageId = packageId == null ? packageEntry.getTitle() : packageId;
                 String packageVersion = packageEntry.getProperties().getVersion().toString();
 
-                NugetHierarchicalArtifactCoordinates c = new NugetHierarchicalArtifactCoordinates(packageId,
+                NugetArtifactCoordinates c = new NugetArtifactCoordinates(packageId,
                         packageVersion,
```
Comment (sbespalov) [inline strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-nuget-layout-provider/src/main/java/org/carlspring/strongbox/repository/NugetRepositoryFeatures.java]: @sanket407 can you also fix the indentation here

**Comment by sanket407** [comment]
@sbespalov @carlspring  fixed !

**Comment by carlspring** [comment]
Thanks! :)

**Pull request merged**